src/T2DMap.cpp

```diff
@@ -900,6 +909,15 @@ inline void T2DMap::drawRoom(QPainter& painter, QFont& roomVNumFont, QPen& pen,
         }
 
     }
+    // Do we need to draw the room name:
+    if (mShowRoomName && areRoomIdsLegible) {
+        painter.save();
+        QColor roomIdColor;
```

Comment:
This isn't used at present and since the colour needed will be the same throughout the whole paint event you'd be marginally faster to set the value for this at the top of the method (as soon as it have been decided to draw a map rather than the error message about not having a map or valid user room number) as I note in the next comment - and then use that colour variable to set the pen colour.